### PR TITLE
Use different build directories when checking out a new SHA.

### PR DESCRIPTION
**Pull request by everestmz**
cc @google/clusterfuzz 

This is a small CL, to seperate build (`/out/clusterfuzz_12345...`) directories not only by testcase but by revision too, in an attempt to keep builds as separate as possible. This could help to ensure we have clean directories when reproducing. 

Since we have the `--current` flag we are building two different revisions in the same out folder right now. This may have some effect on the testcase as we aren't always building from a clean folder.

### CODE REVIEW

File: clusterfuzz/binary_providers.py [hunk 1]
```diff
@@ -122,17 +122,21 @@ def __init__(self, testcase_id, build_url, revision, current, goma_dir,
...
+    _, current_sha = common.execute('git rev-parse HEAD',
+                                    self.source_directory,
+                                    print_output=False)
+    return current_sha.strip()
+
   def out_dir_name(self):
     return os.path.join(self.source_directory, 'out',
-                        'clusterfuzz_' + str(self.testcase_id))
+                        'clusterfuzz_%s_%s' % (str(self.testcase_id),
+                        self.get_current_sha()[:7]))
```
Comment (everestmz) [inline clusterfuzz/binary_providers.py]: I use the first 7 characters of the Sha. I figure if that's good enough for got to do in some cases, it's good enough for us.

File: clusterfuzz/binary_providers.py [hunk 2]
```diff
@@ -122,17 +122,21 @@ def __init__(self, testcase_id, build_url, revision, current, goma_dir,
     self.revision = revision
     self.git_sha = sha_from_revision(self.revision, 'v8/v8')
 
+  def get_current_sha(self):
+    _, current_sha = common.execute('git rev-parse HEAD',
```
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: Use `git rev-parse --short HEAD` and we don't need to do `:7`.

I see that we need it elsewhere. I actually don't mind having a long folder name though.

File: clusterfuzz/binary_providers.py [hunk 3]
```diff
@@ -122,17 +122,21 @@ def __init__(self, testcase_id, build_url, revision, current, goma_dir,
...
+                                    self.source_directory,
+                                    print_output=False)
+    return current_sha.strip()
+
   def out_dir_name(self):
```
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: Let's add a method description explaining that the dirs are unique based on (sha, is_dirty, testcaseid)

File: clusterfuzz/binary_providers.py [hunk 4]
```diff
@@ -122,17 +122,21 @@ def __init__(self, testcase_id, build_url, revision, current, goma_dir,
...
     return os.path.join(self.source_directory, 'out',
-                        'clusterfuzz_' + str(self.testcase_id))
+                        'clusterfuzz_%s_%s' % (str(self.testcase_id),
```
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: Remove `str()`.

File: clusterfuzz/binary_providers.py (hunk 2 above)
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: Let's have the long folder name... so that we don't have to add more test for this method.

**Comment by everestmz** [comment]
Thank you, updated!

**Pull request merged**